homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,56 @@
+"""Coordinator for handling data fetching and updates."""
+
+import logging
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+
+class MadVRCoordinator(DataUpdateCoordinator[dict]):
+    """My custom coordinator for push-based API."""
+
+    config_entry: MadVRConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: MadVRConfigEntry,
```

Comment:
```suggestion
```

after the `super().__init__(` in this file, you can just freely access `self.config_entry`